### PR TITLE
Bourbon update

### PR DESCRIPTION
**Pull request by josephgknox**
# READY FOR REVIEW

# Summary
- Updated to latest version of Bourbon (v5.0.0)
- Updated all deprecated code that we were utilizing (namely `font-face` and Font stack variables)
- Latest version removes prefixing features, so have added autoprefixer and postcss (plus related Grunt command) to add vendor prefixes only where needed (https://github.com/postcss/autoprefixer); also added postcss as part of watch task

# Needed By (Date)
- N/A

# Urgency
- N/A

# Steps to Test

1. Pull this branch
2. Ensure that local version of style guide and pattern library has remained unchanged (visually)
3. Can also test new grunt command (`grunt postcss`); also runs with `grunt watch` if running when change is made

# Affected Projects or Products
- Decanter

# Associated Issues and/or People
- https://github.com/SU-SWS/decanter/issues/112
- https://www.bourbon.io/docs/migrating-from-v4-to-v5/

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)

### CODE REVIEW

File: Gruntfile.js [hunk 1]
```diff
@@ -94,8 +115,10 @@ module.exports = function(grunt) {
   grunt.loadNpmTasks('grunt-contrib-watch');
   grunt.loadNpmTasks('grunt-symlink');
   grunt.loadNpmTasks('grunt-verbosity');
+  grunt.loadNpmTasks('grunt-postcss');
 
   grunt.registerTask('styleguide', ['run:styleguide', 'verbosity:symlinkquiet']);
+  grunt.registerTask('default', ['postcss:dist']);
```
Comment (sherakama) [inline Gruntfile.js]: This is duplicating the default task. Remove this line.

You can, instead, move the `'postcss:dist'` tasks in to the styleguide task array in line 120. You will also probably want to ensure that it is ran after calling `grunt sass` but you cannot simply just add this to the end of that command. As you want to make this a dependency I would suggest registering another a new task called `compile` or something that runs both grunt sass and grunt post-css in the same command. 

eg: `grunt.registerTask('compile', ['sass:dist', 'postcss:dist']);`

File: css/decanter.css [hunk 2]
```diff
@@ -991,7 +991,7 @@ table {
   border-right: 0; }
 
 html {
-  font-family: "Source Sans Pro", "Helvetica Neue", "Helvetica", "Roboto", "Arial", sans-serif;
+  font-family: "Source Sans Pro", "Helvetica Neue", "Helvetica", "Arial", sans-serif;
```
Comment (sherakama) [inline css/decanter.css]: Looks like Roboto went away here. Was this intensional?

File: css/decanter.css [hunk 3]
```diff
@@ -1020,21 +1020,21 @@ h6 {
...
-    font-family: "Source Sans Pro", "Helvetica Neue", "Helvetica", "Roboto", "Arial", sans-serif; }
+    font-family: "Source Sans Pro", "Helvetica Neue", "Helvetica", "Arial", sans-serif; }
   .decanter-serif h1, .decanter-serif
   h2, .decanter-serif
   h3, .decanter-serif
   h4, .decanter-serif
   h5, .decanter-serif
   h6 {
-    font-family: "Crimson Text", "Georgia", "Cambria", "Times New Roman", "Times", serif; }
+    font-family: "Crimson Text", "Georgia", "Times", "Times New Roman", serif; }
```
Comment (sherakama) [inline css/decanter.css]: Cambria is missing here.

File: css/decanter.css [hunk 4]
```diff
@@ -1424,7 +1424,7 @@ a.decanter-button--unstyled {
   .decanter-icon-block .decanter-icon-block__img {
     float: left;
     margin-right: 1rem;
-    margin-right: 1.776889em; }
+    margin-right: 1.5625em; }
```
Comment (sherakama) [inline css/decanter.css]: Was this margin change intensional?

File: package.json [hunk 5]
```diff
@@ -17,21 +17,24 @@
     "url": "https://github.com/SU-SWS/decanter/issues"
   },
   "dependencies": {
-    "bourbon": "^4.2.7",
+    "bourbon": "^5.0.0",
```
Comment (sherakama) [inline package.json]: These are pretty major changes. Please change the version of decanter to `3.0.0`

Comment (josephgknox) [reply package.json]: Any reason why we would want to wait until Neat is updated, too? Or, would that be cause for updating to `4.0.0`?

File: css/decanter.css (hunk 2 above)
Comment (josephgknox) [reply css/decanter.css]: The fonts after `"Source San Pro"` here are generated by the `$font-stack-helvetica`font stack variable. These are fallbacks. Bourbon has updated these.

File: css/decanter.css (hunk 3 above)
Comment (josephgknox) [reply css/decanter.css]: The fonts after `"Crimson Text"` here are generated by the `$font-stack-georgia` font stack variable. These are fallbacks. Bourbon has updated these.

File: css/decanter.css (hunk 4 above)
Comment (josephgknox) [reply css/decanter.css]: Yes. Modular scale is being used on this component (it's the only component utilizing this so far) and it has been updated. We can change it use the margin mixin instead (for now), but there is no real visual difference in the style guide and pattern library as it adjusted relatively.

File: package.json (hunk 5 above)
Comment (kgcreative) [reply package.json]: Neat is a pretty major update as well (By the way, i'm happy to make myself available to answer questions about updating neat. Some of the concepts on how neat handles padding, margins, and containers changed pretty significantly between neat 1.x and the latest stable release (v2.1.0).)

File: Gruntfile.js (hunk 1 above)
Comment (josephgknox) [reply Gruntfile.js]: Good catch! And, I really like the `compile` task idea. I've corrected this and have added that task.

File: package.json (hunk 5 above)
Comment (josephgknox) [reply package.json]: Thanks, @kgcreative! Sounds like another major release with Neat would be a good idea then. So, I'll update the version now and we can again release a new version with the eventual Neat updates.

Thanks for the offer to help; I'll surely want to pick your brain on how to best make the updates.

**Comment by sherakama** [comment]
@JBCSU 

I have reviewed these changes and this PR looks GTG. Please have a review as well. 

The next steps after approval will be:

1. Cut a new `v2` branch off of master (done)
2. Tag and release a version for 2.1.0
3. Merge this PR in to master
4. Tag and release a 3.0.0 version
5. Close out and re-groom the backlog with new versions and goals (we should set a meeting for this.)

File: css/decanter.css (hunk 2 above)
Comment (sherakama) [reply css/decanter.css]: Thanks for confirming.

File: css/decanter.css (hunk 3 above)
Comment (sherakama) [reply css/decanter.css]: Thanks for confirming.

File: css/decanter.css (hunk 4 above)
Comment (sherakama) [reply css/decanter.css]: As long as this change is acceptable to you 👍

File: css/decanter-grid.css [hunk 6]
```diff
@@ -265,6 +266,4 @@ body {
   margin-left: 85.2980433528%; }
 
 .decanter-offset-eleven-twelfths {
-  margin-left: 93.8278476881%; }
-
-/*# sourceMappingURL=decanter-grid.css.map */
```
Comment (sherakama) [inline css/decanter-grid.css]: @josephgknox 
It looks like the source map gets removed with the auto-prefixer. Is this going to cause an error for debugging?

File: Gruntfile.js [hunk 7]
```diff
@@ -37,10 +37,21 @@ module.exports = function(grunt) {
         }
       }
     },
+    postcss: {
+      options: {
+        map: true,
```
Comment (sherakama) [inline Gruntfile.js]: @josephgknox: I have now set this to true in order to preserve the sass maps.

Comment (josephgknox) [reply Gruntfile.js]: @sherakama awesome! I had originally set it to false as the sass task was writing the maps. Based on order of operations (and now that these things happened as part of a single command) this should've been set to true. Thanks!

**Comment by sherakama** [comment]
@josephgknox 

Along with this PR I would like to see an `UPGRADE.md` file with instructions on how to upgrade from 2.x to 3.x

**Comment by josephgknox** [comment]
@sherakama took a first stab at stubbing out an `UPGRADE.md`.

File: UPGRADE.md [hunk 8]
```diff
@@ -0,0 +1,43 @@
+# Upgrade
```
Comment (sherakama) [inline UPGRADE.md]: This is great. Thanks @josephgknox

File: package.json (hunk 5 above)
Comment (kgcreative) [reply package.json]: Neat 2.x (And the upcoming 3.x) can be used as forward-compatible fallbacks for both flexbox and grid. Had the SU Homepage Launch happened 6 months later, I would have used grid for components. CSS grid and flexbox have so many advantages over floated layouts, which will make most float-based frameworks mostly obsolete. Browser support for both are really good. Neat's breakpoint management is really great, but a lot of grid layouts can handle being responsive by using smart fractional units combinations and zero breakpoints, so even that is a massive improvement.

Comment (josephgknox) [reply package.json]: Yes! We do plan on upgrading Neat as the next major undertaking (version `4.0.0`).

Shea and I have also begun to talk about (and want to meet with you, JB, about) adding CSS Grid and/or Flex Box-based classes that can be used in Decanter in addition to the Neat grid. That is, our template files for pages and other items that can and should be more rigid can leverage the current Decanter grid classes (Neat), but patterns and components can leverage more flexible ones (CSS Grid/Flex Box).

In the short-term, though, to get the project in a place where it is utilizing the latest Bourbon and Neat, I thought that this work should be prioritized first, especially before we start building out too many components and patterns that would need to be updated.

**Comment by josephgknox** [comment]
https://github.com/SU-SWS/decanter/tags
https://github.com/SU-SWS/decanter/releases
https://www.npmjs.com/package/decanter

**Pull request merged**